### PR TITLE
Tox, Travis, and Appveyor improvements 

### PR DESCRIPTION
**Pull request by mark-adams**
* Added appveyor support (Thanks @mindw)
* Added tox configuration improvements (Thanks @mindw)
* Added travis configuration improvements (Thanks @mindw)
* Excluded `__main__.py` from coverage

cc: @jpadilla

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -3,66 +3,46 @@
...
-        self.test_suite = True
-
-    def run_tests(self):
-        # import here, cause outside the eggs aren't loaded
-        import pytest
-        errno = pytest.main(['.'] + self.pytest_args)
-        sys.exit(errno)
+from setuptools import find_packages, setup
 
+__version__ = None
```
Comment (jpadilla) [inline setup.py]: Let's change this for what I proposed earlier. [`get_version()`](https://github.com/tomchristie/django-rest-framework/blob/master/setup.py#L11) works well.

Comment (mark-adams) [inline setup.py]: Can you remind me what you proposed?

Comment (jpadilla) [inline setup.py]: Just do the same.

```python
def get_version(package):
    """
    Return package version as listed in `__version__` in `init.py`.
    """
    init_py = open(os.path.join(package, '__init__.py')).read()
    return re.search("__version__ = ['\"]([^'\"]+)['\"]", init_py).group(1)

version = get_version('jwt')
```

**Comment by jpadilla** [comment]
lol I think that's failing because of the é in my last name

**Comment by jpadilla** [comment]
I wonder if this might do.

```python
codecs.open(os.path.join(package, '__init__.py'), 'r', 'utf-8').read()
```

**Comment by mark-adams** [comment]
Coverage won't really go down. Once #159 removes the old Python 3.2-only compatibility code, it'll be back to normal.

What do you think @jpadilla?

**Comment by jpadilla** [comment]
@mark-adams LGTM, all yours, thanks! I'll get the Appveyor account setup.

**Comment by jpadilla** [comment]
This will need rebasing after merging #159

**Comment by jpadilla** [comment]
Repo already set at AppVeyor: https://ci.appveyor.com/project/jpadilla/pyjwt

**Pull request merged**